src/components/Detail/index.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { func, object } from 'prop-types'
+
+import { Header } from '../'
+import GifItem from '../../containers/GifItem'
+import { DetailWrapper, GifWrapper, TitleWrapper } from './Detail.style'
+
+class Detail extends React.Component {
+  componentWillMount() {
+    this.props.initialSetup(this.props.match.params.id)
+  }
+
+  render() {
+    const { giph } = this.props
+
+    return (
+      <div>
+        <Header />
+        <DetailWrapper>
+        { giph !== null &&
```

Comment:
you can save the component in an outside variable, so your validation can seem cleaner